lib/resty/websocket/client.lua

```diff
@@ -91,7 +92,11 @@ function _M.connect(self, uri, opts)
     -- ngx.say("port: ", port)
 
     if not port then
-        port = 80
+        if str_sub(uri, 1, 6) == 'wss://' then
```

Comment:
The `string.sub()` builtin creates a new Lua string, which is a new GC object, which incurs extra GC overhead and dynamic allocation overhead on this potentially hot code path. Better use the `ngx.re.find()` builtin to do the matching without creating any new GC objects.